src/middleware/api.js

```diff
@@ -68,15 +68,21 @@ export default (api) => ({ dispatch, getState }) => (next) => (action) => {
   } else {
     actionPromise = Array.isArray(request)
       ? mode === 'serial'
-        ? request.reduce((prevPromise, item) => {
-            return prevPromise.then(() => {
-              return api[item.op](item.path, {
+        ? request
+            .map((item) =>
+              api[item.op](item.path, {
                 data: item.data,
                 type: item.type,
                 headers: item.headers,
+              }),
+            )
+            .reduce((chain, apiCall) => {
```

Comment:
Here it doesn't waterfall the requests anymore. To be fixed

Reply:
fixed